src/XMakeCommandLine/UnitTests/XMake_Tests.cs

```diff
@@ -1783,11 +1783,32 @@ private string CopyMSBuild()
                     .EnumerateFiles(source)
                     .Where(f=> f.EndsWith(".dll") || f.EndsWith(".tasks") || f.EndsWith(".exe") || f.EndsWith(".exe.config"));
 
+               
+                var directoriesToCopy = Directory
+                    .EnumerateDirectories(source)
+                    .Where(d => Directory.EnumerateFiles(d).Any(f => f.EndsWith("resources.dll")));  // Copy satellite assemblies
+
                 foreach (var file in filesToCopy)
                 {
                     File.Copy(file, Path.Combine(dest, Path.GetFileName(file)));
                 }
 
+                foreach (var directory in directoriesToCopy)
+                {
+                    foreach (var sourceFile in Directory.EnumerateFiles(directory, "*"))
+                    {
+                        var destinationFile = sourceFile.Replace(source, dest);
+
+                        var directoryName = Path.GetDirectoryName(destinationFile);
+                        if (!Directory.Exists(directoryName))
```

Comment:
The usual wisdom is that this is a race condition (the directory could get deleted between this line and the next), so there's no point in checking first--just call `CreateDirectory`.